scalyr_agent/configuration.py

```diff
@@ -153,6 +156,21 @@ def parse(self, logger=None):
             self.__verify_or_set_optional_string(self.__config, 'scalyr_server', 'https://agent.scalyr.com',
                                                  'configuration file %s' % self.__file_path)
 
+            # force https unless otherwise instructed not to
+            if not self.__force_http:
+                server = self.__config['scalyr_server'].strip()
+
+                parts = urlparse.urlparse( server )
+                if not parts.scheme:
+                    https_server = 'https://' + server
+                elif parts.scheme == 'http':
+                    https_server = re.sub( "^http://", "https://", server )
+
+                if https_server != server:
+                    self.__config['scalyr_server'] = https_server
+                    if logger:
+                        logger.info( "Forcing https protocol for server url: %s -> %s.  You can override this with the --force-http flag, but be mindful that there are security implications with doing this, including tramsitting your Scalyr api key over an insecure connection." % (server, self.__config['scalyr_server'] ) )
```

Comment:
If we put this logging statement here, I think it will be emitted everytime we check to see if the configuration file has changed.  Correct?  Maybe instead we just have the Configuration object remember the `raw_server_host` and have a check in the ScalyrClient that sees if the raw host is diferent than the one we are using.. if we so, we emit the warning when we init the ScalyrClient.  Or, we just have a boolean that remembers if we forced it to secure or not.  You get the idea.

Reply:
Ok I've changed this.  The check now happens in the main agent loop.  It needs to happen there because the `ScalyrClient` doesn't have a reference to the config object.

The log message is only output if the server and the raw server are different, *and* the server is different from value the previous time through the loop.

This ensures that the message will be printed once, unless scalyr_server is changed in the config file, and then it will print it again (once) if the value is different from the raw value.